install.sh

```diff
@@ -262,6 +262,60 @@ outdated_glibc() {
   version_lt "$glibc_version" "$REQUIRED_GLIBC_VERSION"
 }
 
+usage() {
+  cat <<EOS
+Homebrew Installer
+Usage: /bin/bash install.sh [--help] [--brew-git-remote URL] [--core-git-remote URL]
+
+This script installs Homebrew, the missing package manager for macOS (or Linux).
+
+Options:
+  -b, --brew-git-remote=URL   Use this URL as the Homebrew/brew git remote.
+                              Default: ${HOMEBREW_BREW_DEFAULT_GIT_REMOTE}.
+
+  -c, --core-git-remote=URL   Use this URL as the Homebrew/homebrew-core git remote.
+                              Default: ${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}.
+
+  -h, --help                  Show this help message and exit.
+EOS
+  exit "${1:-0}"
+}
+
+# Parse remote URLs of Homebrew repositories from command line arguments if presented.
+parse_remote_urls_from_arguments() {
+  while [[ "$#" -gt 0 ]]; do
+    opt="$1"
+    shift  # expose the next argument
+    case "$opt" in
+      --) break 2 ;;
+      -b | --brew-git-remote)
+        HOMEBREW_BREW_GIT_REMOTE="${1:-}"; shift ;;
+      --brew-git-remote=*)  # alternative format: --brew-git-remote=URL
+        HOMEBREW_BREW_GIT_REMOTE="${opt#*=}" ;;
+      -c | --core-git-remote)
+        HOMEBREW_CORE_GIT_REMOTE="${1:-}"; shift ;;
+      --core-git-remote=*)  # alternative format: --core-git-remote=URL
+        HOMEBREW_CORE_GIT_REMOTE="${opt#*=}" ;;
+      -h | --help)
+        usage ;;
+      *)
+        warn "Unrecognized option: '${opt}'"; usage 1;;
+    esac
+  done
+}
+
+unset HOMEBREW_{BREW,CORE}_GIT_REMOTE  # unset variables from environment
+if [[ "$(basename -- "$0")" != "bash" && "$(basename -- "$0")" != "install.sh" ]]; then
+  # The script is invoked by `bash -c "$(cat install.sh)" [--brew-git-remote URL] [--core-git-remote URL]`
+  # Unshift the first argument back to the argument list
+  parse_remote_urls_from_arguments "$0" "$@"
```

Comment:
you also could `set -- "$0" "$@"` here instead and then use `parse_remote_urls_from_arguments "$@"` universally (outside of if/else).